### PR TITLE
6 new files: 1 .txt with pseudocode for problems and 5 .py files to te…

### PR DESCRIPTION
**Pull request by audreycode6**
…st pseudocode and make program work

# Description
Practicing writing formal pseudocode before coding a program. pseudocode.txt holds the program requirements and then my pseudocode. test1-test5.py has each problems pseudocode with the real code, so each test#.py has a working function/program to match psedocode/problem requirements from pseudocode.txt.

# Changes
New .txt and .py files added to L2_PY101_LS repo.

# QA Instructions

1. Check pseudocode.txt and give feedback on pseudocode.
2. Ensure the test#.py files works as is intended.

### CODE REVIEW

File: lesson_2/test5.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # SET/READ new_list lists merged as list of tuples
+    new_list = list(zip_list)
+    # SET merge_list = to append values of tuple
+    merge_list = []
+    # SET tuple1, tuple2 to store values from in tuple
+    tuple1 = 0
+    tuple2 = 0
+    # WHILE for items in new_list index items from tuple and append to merge_list
+    for element in new_list:
+        tuple1 = element[0]
```
Comment (Doxify) [inline lesson_2/test5.py:21]: You don't need to explicitly create variables, you can loop through a list of tuples like this:

```suggestion

    # WHILE for items in new_list index items from tuple and append to merge_list
    for element1, element2 in new_list:
        merge_list.append(element1)
        merge_list.append(element2)
```

File: lesson_2/test1.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+# 1. a function that returns the sum of two numbers
+
+# START
+# GET 2 numbers from user
```
Comment (Doxify) [inline lesson_2/test1.py:5]: Here's how I would organize this code:

```suggestion
def sum(num1, num2):
    return num1 + num2

num1 = int(input('Enter a number: '))
num2 = int(input('Enter another number: '))

print(sum(num1, num2))
```

File: lesson_2/test2.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# 2. a function that takes a list of strings, and returns a string that is all those strings 
+# concatenated together
+
+# START
+# GET list of strings
+list = ['hi', 'bye', 'haha', 'fuck']
+
+def add_strings(list):
+    # SET length: len(list) + 1
+    length = len(list) + 1
```
Comment (Doxify) [inline lesson_2/test2.py:11]: nit: this variable (`length`) is never used

File: lesson_2/test3.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# # => [1,7,5]
+
+# START
+# GET list_ints: list of ints 
+list = [1, 2, 3, 4]
+def every_other_element(list):
+# SET: new_list : []
+    new_list = []
+# WHILE: for list_element in range(::2) -->new_list.append(list_element)
+    for element in range(1, len(list) + 1, 2):
```
Comment (Doxify) [inline lesson_2/test3.py:13]: shouldn't this be `range(0, len(list) + 1, 2)`? It says to start at the first element but you're starting at the second

**Review by Doxify — APPROVED** [review]
looks fine, left some feedback

**Comment by Doxify** [comment]
also, this pr title is not good

File: lesson_2/test5.py (hunk 1 above)
Comment (audreycode6) [reply lesson_2/test5.py:21]: good to know

File: lesson_2/test1.py (hunk 2 above)
Comment (audreycode6) [reply lesson_2/test1.py:5]: fixed it but why have have num1 and num2 intialized below? is it bc we are calling function below so its just cleaner to have it above  the call? (ie line 9 )//also I wont leave comments like that in future ik its ugly but wanted to try it out so it code be pseudocode step and then matching code

File: lesson_2/test2.py (hunk 3 above)
Comment (audreycode6) [reply lesson_2/test2.py:11]: and I oopt- good call out

File: lesson_2/test3.py (hunk 4 above)
Comment (audreycode6) [reply lesson_2/test3.py:13]: woah very true; I tested with diff input and actually whole program wasn't working --> just fixed will do another PR

**Pull request merged**